### PR TITLE
Added permission node for Nickname Prefix bypass

### PR DESCRIPTION
**Pull request by Baylem**
### Information

This PR closes #4994. 

### Details

**Proposed feature:**  

Allow a permission to determine whether the nickname prefix gets applied to a group or user. It also removes a statement that is not true in the essentials config, the nickname prefix is NOT included in the nickname max length.

**Environments tested:**    

OS: Windows 11

Java version:  17.0.2

- [x] Most recent Paper version (1.19, git-Paper-61)
- [x] CraftBukkit/Spigot/Paper 1.12.2
- [ ] CraftBukkit 1.8.8

**Demonstration:**  

Before change: https://i.imgur.com/HZaKazT.png
After change: https://i.imgur.com/QyitAPl.png

### CODE REVIEW

**Comment by Zelathis** [comment]
I just noticed that I did not commit the plugin.yml, this will need the permission added for default op (if necessary)

**Review by pop4959 — DISMISSED** [review]

**Review by mdcfe — CHANGES_REQUESTED** [review]
Thanks for the PR! I think `essentials.nick.prefix.hide` is a bit clearer of a name for the permission. Otherwise, I like this feature and would be happy to merge it if you're happy with changing the permission name.

**Comment by Evidentsinger14** [comment]
> Thanks for the PR! I think `essentials.nick.prefix.hide` is a bit clearer of a name for the permission. Otherwise, I like this feature and would be happy to merge it if you're happy with changing the permission name.

I feel like `essentials.nick.hideprefix` would be better - there aren’t any other permissions that have follow .nick.prefix (that I’m aware of).

**Review by Baylem — COMMENTED** [review]
I have updated the permission node with Evident's suggestion

File: Essentials/src/main/resources/config.yml [hunk 1]
```diff
@@ -27,9 +27,10 @@
 ops-name-color: '4'
 
 # The character(s) to prefix all nicknames, so that you know they are not true usernames.
+# Users with essentials.nick.prefix.bypass will not be prefixed with the character(s)
```
Comment (mdcfe) [inline Essentials/src/main/resources/config.yml]:
```suggestion
# Users with essentials.nick.hideprefix will not be prefixed with the character(s)
```

File: Essentials/src/main/java/com/earth2me/essentials/User.java [hunk 2]
```diff
@@ -464,7 +464,7 @@ public String getNick(final boolean withPrefix, final boolean withSuffix) {
         } else if (nick.equalsIgnoreCase(getName())) {
             nickname = nick;
         } else {
-            nickname = FormatUtil.replaceFormat(ess.getSettings().getNicknamePrefix()) + nick;
+            nickname = isAuthorized("essentials.nick.hideprefix") ? nick : FormatUtil.replaceFormat(ess.getSettings().getNicknamePrefix()) + nick;
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/User.java]:
```suggestion
            if (isAuthorized("essentials.nick.hideprefix")) {
                nickname = nick;
            } else {
                nickname = FormatUtil.replaceFormat(ess.getSettings().getNicknamePrefix()) + nick;
            }
```

**Review by mdcfe — APPROVED** [review]

**Pull request merged**